Para investigar se existe uma associação positiva entre as notas obtidas em Cálculo e as classificações em Estatística, vamos usar o coeficiente de correlação de Spearman. Este é um teste não paramétrico usado para medir a força e a direção de associação entre duas variáveis que são medidas em uma escala ordinal.

As categorias de Estatística, "Excelente", "Muito Bom", "Bom", "Suficiente", "Insuficiente", "Mau", e as notas de Cálculo, "A", "B", "C", "D", "E", "F", podem ser convertidas em escalas numéricas ordinais para análise. Por exemplo, podemos atribuir valores de 6 a 1, respectivamente, para as classificações de Estatística e valores de 6 a 1 para as notas de Cálculo, de "A" a "F".

Com esses dados codificados, podemos calcular o coeficiente de correlação de Spearman. Vou começar atribuindo os valores numéricos para as categorias e notas e depois calcular o coeficiente de correlação.

In [1]:
from scipy.stats import spearmanr

# Dados das notas codificados numericamente
# Estatística: Excelente (6), Muito Bom (5), Bom (4), Suficiente (3), Insuficiente (2), Mau (1)
# Cálculo: A (6), B (5), C (4), D (3), E (2), F (1)

# Dados codificados
estatistica = [6, 6, 6, 5, 5, 5, 5, 4, 4, 4, 4, 4, 4, 3, 3, 3, 2, 2, 2, 2, 1, 1, 1, 1, 1]
calculo = [6, 6, 4, 4, 4, 3, 2, 6, 5, 4, 4, 3, 3, 5, 5, 5, 4, 4, 2, 1, 3, 2, 3, 3, 3]

# Cálculo do coeficiente de correlação de Spearman
coeficiente_spearman, p_value = spearmanr(estatistica, calculo)

coeficiente_spearman, p_value


(0.4546021025901462, 0.02242600579752975)

O coeficiente de correlação de Spearman é aproximadamente 0.455, e o valor-p associado é aproximadamente 0.0224.

Dado que o valor-p é menor do que o nível de significância de 5% (α=0.05), há evidências estatísticas suficientes para rejeitar a hipótese nula de que não há associação entre as notas. 
Isso indica que existe uma correlação positiva moderada entre as classificações obtidas em Estatística e as notas de Cálculo. 
As classificações mais altas em Estatística tendem a estar associadas a notas mais altas em Cálculo.